packages/extension/src/components/pages/History/History.tsx

```diff
@@ -1,25 +1,64 @@
 import { FC, useEffect, useState } from 'react';
 
-import { CircularProgress } from '@mui/material';
+import { Button, CircularProgress, Typography } from '@mui/material';
+import * as Sentry from '@sentry/react';
 
-import { useLedger } from '../../../contexts';
+import { useLedger, useNetwork } from '../../../contexts';
 import { AccountTransaction } from '../../../types';
+import { InformationMessage } from '../../molecules';
 import { TransactionListing } from '../../organisms';
 import { PageWithHeader } from '../../templates';
 
 export const History: FC = () => {
   const { getTransactions } = useLedger();
+  const { reconnectToNetwork } = useNetwork();
 
   const [transactions, setTransactions] = useState<AccountTransaction[] | null>(null);
+  const [isTxFailed, setIsTxFailed] = useState<boolean>(false);
 
   useEffect(() => {
     const getTx = async () => {
-      const tx = await getTransactions();
-      setTransactions(tx);
+      try {
+        const tx = await getTransactions();
+        setTransactions(tx);
+      } catch (e) {
+        if ((e as Error).message === 'You need to be connected to a ledger to make a transaction') {
```

Comment:
Ideally I would extract the error message into a const in the ledgercontext to prevent breaking changes if we change it

Reply:
Good point, pushing that change